src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_ARRAY = 1.5;
+    private int size;
+    private T[] elementData;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            growIfArrayFull();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new
+                    ArrayListIndexOutOfBoundsException("the given index"
+                    + index
+                    + " does not exist");
+        }
+        if (size >= elementData.length) {
+            growIfArrayFull();
+        }
+        //shift of positions to the right side
+        for (int i = size; i > index; i--) {
+            elementData[i] = elementData[i - 1];
```

Comment:
Use System.arraycopy() to move your array elements instead of manual loop copying.